api/types/src/error.rs

```diff
@@ -93,6 +93,8 @@ pub enum AptosErrorCode {
     SequenceNumberTooOld = 402,
     /// The submitted transaction failed VM checks.
     VmError = 403,
+    /// The submitted transaction doesn't have a valid expiration time.
+    InvalidExpirationTime = 404,
```

Comment:
Yes, it's fine to add a new expiration time error code, since it doesn't conflict with the current codes.  Any downstream code should handle unknown codes (though you should check that the Rust rest client does)